agents1/GenericAgent.py

```diff
@@ -442,39 +434,54 @@ def decide_on_bw4t_action(self, state: State):
         return res
 
     def _sendMessage(self, msg):
-        '''
+        """
         Enable sending messages in one line of code
-        '''
+        """
         if msg is None:
             return
 
         if msg.content not in self.received_messages:
             self.send_message(msg)
 
     def _processMessages(self, teamMembers):
-        '''
+        """
         Process incoming messages and create a dictionary with received messages from each team member.
-        '''
+        """
         receivedMessages = {}
         for member in teamMembers:
             receivedMessages[member] = []
 
         while len(self.received_messages) != 0:
             msg = self.received_messages.pop(0)
+            msg = MessageBuilder.process_message(msg)
 
             for member in teamMembers:
-                if msg.from_id == member:
-                    # todo: update goal block location only if its location is closer than the one we have
+                if msg['from_id'] == member:
+                    # TODO: now, the agent assumes all messages can be trusted
                     # todo: update only if you trust the agent
-                    content = MessageBuilder.process_message(msg)           # process message
-                    receivedMessages[member].append(content)
+                    # update goal block location
+                    if msg['type'] is MessageType.FOUND_GOAL_BLOCK:
+                        # find the goal block
+                        for key, goal_block in self._goal_blocks.items():
+                            if goal_block["visualization"] == msg['visualization']:
+                                self.update_goal_block(key, msg['location'], None)
+
+                    elif msg['type'] is MessageType.MOVE_TO_ROOM \
```

Comment:
do not think it makes sense to add the room to `com_visited_rooms` if the agent is saying it's moving towards it

Reply:
yeah, true. I thought it would make sense for the agent to know where other agents are heading and pick a different room to search through

Reply:
This might also have a separate variable, which would also allow us to check the messages with type `OPEN_DOOR` and see if the agent who said it is moving there opened it (or not)

Reply:
> This might also have a separate variable, which would also allow us to check the messages with type `OPEN_DOOR` and see if the agent who said it is moving there opened it (or not)

this should be doable in the trust system, the `receivedMessages` dict stores the messages per agent